api/parts/mgmt/cms.js

```diff
@@ -0,0 +1,206 @@
+/**
+* This module is meant for handling CMS API requests.
+* @module api/parts/mgmt/cms
+*/
+
+/** @lends module:api/parts/mgmt/cms */
+var cmsApi = {},
+    common = require('./../../utils/common.js');
+
+const AVAILABLE_API_IDS = ["server-guides", "server-consents", "server-intro-video", "server-quick-start"],
+    token = "17fa74a2b4b1524e57e8790250f89f44f364fe567f13f4dbef02ef583e70dcdb700f87a6122212bb01ca6a14a8d4b85dc314296f71681988993c013ed2f6305b57b251af723830ea2aa180fc689af1052dd74bc3f4b9b35e5674d4214a8c79695face42057424f0494631679922a3bdaeb780b522bb025dfaea8d7d56a857dba",
+    baseURL = "https://cms.count.ly/api/";
+
+
+/**
+* Get entries for a given API ID from Countly CMS
+* @param {params} params - params object
+* @param {function} callback - callback function
+**/
+function fetchFromCMS(params, callback) {
+    const url = baseURL + params.qstring._id;
+    const pageSize = 100;
+    var results = [];
+
+    /**
+    * Get a single page of data
+    * @param {number} pageNumber - page number
+    **/
+    function fetchPage(pageNumber) {
+        const pageUrl = `${url}?pagination[page]=${pageNumber}&pagination[pageSize]=${pageSize}`;
+
+        fetch(pageUrl, {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': 'Bearer ' + token
+            }
+        })
+            .then(response => response.json())
+            .then(responseData => {
+                const {data, meta} = responseData;
+                if (data && data.length > 0) {
+                    // Add data to results
+                    results = results.concat(data);
+                }
+
+                if (meta && meta.pagination && meta.pagination.page < meta.pagination.pageCount) {
+                    // Fetch next page
+                    fetchPage(meta.pagination.page + 1);
+                }
+                else {
+                    // All pages fetched or no pagination metadata, invoke callback
+                    callback(null, results);
+                }
+            })
+            .catch(error => {
+                console.log(error);
+                callback(error, null);
+            });
+    }
+
+    fetchPage(1);
+}
+
+/**
+* Transform and store CMS entries in DB
+* @param {params} params - params object
+* @param {array} data - data array
+* @param {function} callback - callback function
+**/
+function transformAndStoreData(params, data, callback) {
+    let lu = Date.now();
+    if (!data || data.length === 0) {
```

Comment:
Can we do this deletion in a bit different way?
Do at first bulk update for each:
https://github.com/Countly/countly-enterprise-plugins/blob/master/plugins/drill/api/parts/data/drill.js#L2791-L2797

Then delete  where {'_id': {'$regex': `^${params.qstring._id}`} && lu <current value.